### ISSUE TITLE
Avatars are automatically considered to be non-presentational

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris#1925]
## Issue summary

Currently, the avatar component renders a presentational `img` that may or may not have meaning that should be conveyed to screen readers. The `img` is wrapped in a span with a role of `image` and an `aria-label` is generated even when an `accessibilityLabel` is not set. This is a problem when an avatar is used for visual differentiation only and does not convey additional meaning. 

In the case of the Users list view, the `name` set on the `Avatar` is set as the `aria-label` on the span, so when a merchant using a screen reader goes through the content on the page, the name of each user in the resource list is read twice.

<img width="416" alt="Screen Shot 2019-08-06 at 14 06 56" src="https://user-images.githubusercontent.com/18447883/62564940-c77a7b00-b853-11e9-8b38-512d781073ac.png">

<img width="1068" alt="Screen Shot 2019-08-06 at 14 07 49" src="https://user-images.githubusercontent.com/18447883/62564955-cea18900-b853-11e9-9c60-2897a66acb92.png">

## Expected behavior

If an `accessibilityLabel` is not set on the `Avatar`, I don't think we should assume the image is conveying meaning that needs to be labelled.

## Actual behavior

`Avatar` always sets an `aria-label` on the `span` it renders, so there's no way to avoid redundant screen reader content.

## Steps to reproduce the problem

1. Check out the examples of the[ ResourceList](https://polaris.shopify.com/components/lists-and-tables/resource-list) in the style guide
2. Click Command + F5 to activate Voiceover and hear how a screen reader experiences a resource list item being read that has an `Avatar` labelled with the same content as is read immediately before the text content of the item.

**Comment by danbrady** [comment]
Hi! I looked into this issue. Here's what I found:

Currently, if you do not provide a value or use an empty string for `name`, `initials` or `accessibilityLabel`, the `aria-label` is the i18n string "Avatar". Below is a rendering of `<Avatar />`:

![image](https://user-images.githubusercontent.com/381190/230992310-48f1e7c0-3b37-466c-ba1e-46fb76f4d306.png)


(Same thing happens if `source` added.)

If we suppress that "Avatar" string but keep the `role="img"`, we will get an a11y error because `role="img"` must have _non-empty_ alternative text provided by `aria-label`, `aria-labelledby`, or `title`. ([Reference](https://dequeuniversity.com/rules/axe/4.1/role-img-alt))


There are a few potential changes for presentational avatars when no alternative text is provided. We can:

1. Remove the `role="img"` on the wrapping span if no label is provided.
2. Change the `role="img"` to be `role="presentation"` instead.
3. Others?

**Comment by chloerice** [comment]
Thanks for digging into this @danbrady! I think option 2 makes the most sense if they're purely visual 💯  They shouldn't be in the Accessibility tree unless they add clarity or meaning to adjacent content.
